### DISCUSSION TITLE
Hue Motion Sensor reports out of date illuminance when motion detected

### DISCUSSION THREAD

**Discussion by p-a-dev** [Koenkk/zigbee2mqtt#9326] [Devices]
I'm having some issues with a new Hue Motion Sensor (model 9290012607, updated to latest firmware from 20190219). I'm using the sensor and Node-Red to build a night-light (i.e. turn on lights when motion is detected and the room is dark).

The main issue is that when motion is detected, the immediate published message contains old illuminance readings, usually followed up to 5 seconds later with a new message containing updated illuminance:

> MQTT publish: topic 'zigbee2mqtt/hueMotion01', payload '{"battery":100,"illuminance":0,"illuminance_lux":0,"last_seen":"2021-10-25T14:43:23+01:00","led_indication":true,"linkquality":0,"motion_sensitivity":"high","occupancy":true,"occupancy_timeout":0,"temperature":18.55,"update":{"state":"idle"},"update_available":false}'
> MQTT publish: topic 'zigbee2mqtt/hueMotion01', payload '{"battery":100,"illuminance":24895,"illuminance_lux":309,"last_seen":"2021-10-25T14:43:23+01:00","led_indication":true,"linkquality":0,"motion_sensitivity":"high","occupancy":true,"occupancy_timeout":0,"temperature":18.55,"update":{"state":"idle"},"update_available":false}'

In the example above, I turned on the lights in the room before I went in, so it was very bright.  BUT, the 1st message shows the room is pitch black.  This is followed immediately by a message saying the room is bright.

When a message from the sensor comes in, how can I tell how old each of the individual data points is?  Can I somehow limit reporting to only parameters that are current and accurate?  Or will the messages always contain the entire state of the sensor using last known data (no matter how old the readings are)?

**Comment by sjorge** [comment]
```
 # Optional: state caching, MQTT message payload will contain all attributes, not only changed ones.
  # Has to be true when integrating via Home Assistant (default: true)
  cache_state: true
  # Optional: persist cached state, only used when cache_state: true (default: true)
  cache_state_persistent: true
  # Optional: send cached state on startup, only used when cache_state_persistent: true (default: true)
  cache_state_send_on_startup: true
  # Optional: Logging level, options: debug, info, warn, error (default: info)
```
--- https://www.zigbee2mqtt.io/information/configuration.html

You can play with the cache options if you don't want to include the entire cached state, for me the motion and luminance data arrave in 2 zigbee message so I have set a debounce of 0.5 for my hue motion sensors.

But of course once the light is turned on, the sensor will detect more light and send another message. But at least the info in the message I use to determin if the light needs to turn on includes the currently active luminance data too.

**Reply by p-a-dev** [reply]
Thanks @sjorge! That's a big help.

Do you know if it's possible to set cache_state on a device-by-device basis, or only globally for the system?

**Reply by sjorge** [reply]
They are not listed on the per device options, so i’m pretty sure that’s a no 

https://www.zigbee2mqtt.io/information/configuration.html#device-specific-configuration
